confgenerator/fluentbit/modular.go

```diff
@@ -26,6 +26,9 @@ type Component struct {
 	Kind string
 	// Config is a set of key-value configuration pairs
 	Config map[string]string
+	// RepeatedConfig is used for configuration pairs where the
+	// key can appear in the output fluent bit config multiple times
+	RepeatedConfig map[string][]string
```

Comment:
+1, this is a great observation. `[][2]string` even has the advantage of being fewer characters :P

Merging them to one field would mean that we lose the current behavior of sorting the keys. That might make some of the current uses harder, because items would have to be added in order (and accessing items in `Config` would be harder if the caller wants to mutate them), but I'm 50/50 on whether it's a net win to merge them into one.

Reply:
@jefferbrecht I was working on the change you recommended, and it seemed cleaner to do it as a separate PR (https://github.com/GoogleCloudPlatform/ops-agent/pull/231). The change is really all in this commit https://github.com/GoogleCloudPlatform/ops-agent/pull/231/commits/4c60f04f6ef40abacd23bb9b1a67ded5c4b142a6 but it changed enough things across the fluent bit code base that it felt cleaner to do it separately.

Reply:
Regarding sorting, can we sort by the key then value? Either way, we could continue this discussion in the follow up PR. I left a comment about it in https://github.com/GoogleCloudPlatform/ops-agent/pull/231#discussion_r723785292.